Develop #9 HMDA summary data

rpitingolo opened the pull request:
@ptatian can you please review? There are 3 main components in this pull request review:

1. %Gender, %Prop_type, %purp_type and %race_calc04 were all copied over from the NatData programming. I had to edit the race macro to get it working. The others are as-is.

2. %hmda_tracts_new is a macro that first creates variables for the DC data warehouse and then creates a tract-level summary file for each year of data (2009-2011 in 2000 tracts, 2012 and newer in 2010 tracts). We can add more variables to this as needed, but for now can you review to make sure what's in there looks correct?

3. The main program HMDA_sum_P09.sas takes the tract-level files and merges them together (one for the 2000 tracts, another for the 2010 tracts) and then runs %Create_all_summary_from_tracts to create two separate summary files. Then at the end of the program it merges those two summary files together into a final summary file.

Please let me know if you have any questions.

rpitingolo committed:
Summary file for 2009-2011

rpitingolo committed:
summary program updates

rpitingolo committed:
hmda macros from nat data

rpitingolo committed:
recode race values 6 through 8

rpitingolo committed:
move var creation code to macro

rpitingolo committed:
rename main program to hmda_sum_p09

rpitingolo committed:
bugfixes

rpitingolo committed:
add median variables to summary

ptatian commented:
Thanks, @rpitingolo. I will take a look.

ptatian commented:
@rpitingolo Is there any documentation for the current HMDA source data? If so, can you please put [on Box](https://urbanorg.app.box.com/folder/5520145501)?

rpitingolo commented:
@ptatian when you say source data do you mean the Loans_yyyy files that are in the warehouse already? Or the raw files that creates them?

ptatian commented:
@rpitingolo The raw files. You have some notes about codes for race, etc. so I want to be able to check that. Thanks.

rpitingolo commented:
@ptatian sure. I just saved the raw data values here: https://urbanorg.app.box.com/file/363949575901

ptatian commented on Macros/hmda_tracts_new.sas:
```diff
@@ -0,0 +1,273 @@
+
+%macro hmda_tracts_new (start_yr,end_yr);
+
+%do year = &start_yr. %to &end_yr.;
```
```suggestion
%local year geo geovfmt count_list median_list;

%do year = &start_yr. %to &end_yr.;
```

ptatian replied:
You should always declare local macro variables to avoid conflicts outside the macro.

ptatian commented on Macros/Gender.sas:
```diff
@@ -0,0 +1,88 @@
+ 
+ 
+ 
+ %mend;
```
Best practice is to include macro name in %mend statement.

ptatian commented on Macros/hmda_tracts_new.sas:
```diff
@@ -0,0 +1,273 @@
+
+%end;
+
+%mend hmda_yr;
```
Macro name on %mend is not correct.

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+  calc_vars_labels=,
+  tract_yr=2000,
+  register=,
+  finalize=y,
```
Do we want to save these as permanent data sets? It seems like it is sufficient to keep only the final summary files. In which case, this should be _finalize=n_.  

But, now I'm realizing this parameter no longer does anything. So, we'll have to fix the %Create_summary_from_tracts() macro in the General library. I've created a [new issue](https://github.com/NeighborhoodInfoDC/General/issues/67) for this.

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+  calc_vars_labels=,
+  tract_yr=2010,
+  register=,
+  finalize=y,
```
Same here: _finalize=n_.

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+/* Merge summary files together into one big summary file */
+%macro hmda_merge(geo,geosuf);
+
+data hmda_p09_sum_&geosuf.;
```
When we've done something similar elsewhere (like in NCDB), we've put the modifier _after_ \_sum in the file name: **hmda_sum_p09_&geosuf.**

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+
+%mend hmda_merge;
+%hmda_merge (geo2000,tr00);
+%hmda_merge (geo2010,tr10);
```
There should be 178 tracts in the DC 2010 tract summary, but the output data set has 180.

rpitingolo replied:
@ptatian I think there are 179  tracts in 2010? At least that's what's in the General.Geo2010 dataset.

ptatian replied:
@rpitingolo Yes, you are right. 179 tracts.

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+data vars2000p;
+	set vars2000;
+	prefix = upcase(substr(name,1,3));
+run;
```
Loan count variables with missing values should be recoded to 0, since they represent no lending activity.

ptatian commented on Prog/HMDA_sum_P09.sas:
```diff
@@ -0,0 +1,174 @@
+data hmda_long10_tr10;
+	merge &tr2010data.;
+	by geo2010;
+run;
```
Loan count variables with missing values should be recoded to 0, since they represent no lending activity.

ptatian commented on Macros/hmda_tracts_new.sas:
```diff
@@ -0,0 +1,273 @@
+	/* Create income categories from ipums %hud_inc macro */
+	if hhincome ^= . then do;
+	numprec = 4;
+	%Hud_inc_2009;
```
It does not seem like we should always use the 2009 income macro. We should use the macro that corresponds to the HMDA data year.

ptatian reviewed (changes requested):
Hi @rpitingolo. This is 90% there. Just need a few adjustments. Let me know if you have any questions.

rpitingolo commented:
@ptatian thanks. I looked at the suggested revisions and everything seems clear. I'll let you know if I have questions as I'm editing.

rpitingolo committed:
declare local macros

rpitingolo committed:
add macro name to %mend

rpitingolo committed:
finalize=n for geo creation files

These macro convert tract-level data to all dc geographies, we don't need these saved  permanently.

rpitingolo committed:
switch p09 after the _sum_

rpitingolo committed:
erroneous tract on geo2010 file

rpitingolo committed:
Loan count variables with missing values recoded to 0

rpitingolo committed:
make sure hud_inc_yyyy runs for correct year

rpitingolo committed:
naming cleanup
